cmd/fs-v1.go

```diff
@@ -666,8 +676,9 @@ func (fs *fsObjects) PutObject(bucket string, object string, data *hash.Reader,
 // putObject - wrapper for PutObject
 func (fs *fsObjects) putObject(bucket string, object string, data *hash.Reader, metadata map[string]string) (objInfo ObjectInfo, retErr error) {
 	// No metadata is set, allocate a new one.
-	if metadata == nil {
-		metadata = make(map[string]string)
+	meta := make(map[string]string)
```

Comment:
what is the need for this copy?

Reply:
To avoid concurrent map read and write panic  - fs putObject modifies the metadata map by inserting ETag into the map while cache fs ranges over the k-v in the metadata and copies the entries.